DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/src/ac_fixed.cpp

```diff
@@ -0,0 +1,200 @@
+// clang-format off
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/ac_types/ac_fixed.hpp>
+#include <sycl/ext/intel/ac_types/ac_fixed_math.hpp>
+// clang-format on
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+using fixed_10_3_t = ac_fixed<10, 3, true>;
+using fixed_9_2_t = ac_fixed<9, 2, true>;
+
+// Forward declare the kernel name in the global scope.
+// This is a FPGA best practice that reduces name mangling in the reports.
+class ConstructFromFloat;
+class ConstructFromACFixed;
+class CalculateWithFloat;
+class CalculateWithACFixed;
+
+// Not recommended Usage example:
+// Convert dynamic float value inside the kernel
+void TestConstructFromFloat(queue &q, const float &a,
+                            ac_fixed<20, 10, true, AC_RND, AC_SAT> &b) {
+  buffer<float, 1> inp_buffer(&a, 1);
+  buffer<ac_fixed<20, 10, true, AC_RND, AC_SAT>, 1> ret_buffer(&b, 1);
+
+  q.submit([&](handler &h) {
+    accessor in_acc{inp_buffer, h, read_only};
+    accessor out_acc{ret_buffer, h, write_only, no_init};
+
+    h.single_task<ConstructFromFloat>([=] {
+      ac_fixed<20, 10, true, AC_RND, AC_SAT> t(in_acc[0]);
+      ac_fixed<20, 10, true, AC_RND, AC_SAT> some_offset(0.5f);
+      out_acc[0] = t + some_offset;
+    });
+  });
+}
+
+// Recommended Usage example:
+// Convert dynamic float value outside the kernel
+void TestConstructFromACFixed(queue &q,
+                              const ac_fixed<20, 10, true, AC_RND, AC_SAT> &a,
+                              ac_fixed<20, 10, true, AC_RND, AC_SAT> &ret) {
+  buffer<ac_fixed<20, 10, true, AC_RND, AC_SAT>, 1> inp_buffer(&a, 1);
+  buffer<ac_fixed<20, 10, true, AC_RND, AC_SAT>, 1> ret_buffer(&ret, 1);
+
+  q.submit([&](handler &h) {
+    accessor in_acc{inp_buffer, h, read_only};
+    accessor out_acc{ret_buffer, h, write_only};
+
+    h.single_task<ConstructFromACFixed>([=] {
+      ac_fixed<20, 10, true, AC_RND, AC_SAT> t(in_acc[0]);
+      ac_fixed<20, 10, true, AC_RND, AC_SAT> some_offset(0.5f);
+      out_acc[0] = t + some_offset;
+    });
+  });
+}
+
+void TestCalculateWithFloat(queue &q, const float &x, float &ret) {
+  buffer<float, 1> inp_buffer(&x, 1);
+  buffer<float, 1> ret_buffer(&ret, 1);
+
+  q.submit([&](handler &h) {
+    accessor x{inp_buffer, h, read_only};
+    accessor res{ret_buffer, h, write_only, no_init};
+
+    h.single_task<CalculateWithFloat>([=] {
+      float sin_ret = sinf(x[0]);
```

Comment:
For better readability would using sin_x (instead of sin_ret) and cos_x (instead of cos_ret) work better?